source/npm/util.js

```diff
@@ -107,3 +109,74 @@ exports.checkIgnoreStrategy = ({files}) => {
 		`);
 	}
 };
+
+// New files added part of .npmignore or not part of files-attribute in "package.json"
+exports.checkNewFiles = async (newFiles, filesFromPackageDotJson) => {
+	const rootDir = pkgDir.sync();
+	const npmignoreExists = fs.existsSync(path.resolve(rootDir, '.npmignore'));
+
+	let result = [];
+
+	if (filesFromPackageDotJson) {
+		result = getFilesNotPartOfFileAttribute(filesFromPackageDotJson, newFiles);
+	}
+
+	if (npmignoreExists) {
+		result = result.concat(await getFilesIgnoredByDotnpmignore(newFiles));
+	}
+
+	return result;
+};
+
+async function getFilesIgnoredByDotnpmignore(fileList) {
+	if (!Array.isArray(fileList)) {
+		throw new TypeError('expected array, but got {typeof fileList}');
+	}
+
+	const result = [];
+	await ignoreWalker({
+		path: pkgDir.sync(),
+		ignoreFiles: ['.npmignore']
+	})
+		.then(WhiteList => {
+			if (!Array.isArray(WhiteList)) {
+				throw new TypeError(`Expected array, but got ${typeof WhiteList} `);
+			}
+
+			// Every file in fileList should be part of the WhiteList
+			fileList.forEach(item => {
+				const found = WhiteList.find(WhiteListItem => {
+					return WhiteListItem === item;
+				});
+				if (found === undefined) {
+					result.push(item);
+				}
+			});
```

Comment:
Prefer `for-of` everywhere (not just in this instance)

Also this can probably be a single loop:

```js
for (const file of fileList) {
	if (WhiteListItem === file) {
		result.push(item);
		break;
	}
}
```

Reply:
I put it in ```for-of``` loop.

Reply:
The find function is not necessary. It’s just a loop. Adjust the condition and you can drop .find

Reply:
The condition should be `array.includes(file)`

Reply:
With my latest push i implemented your suggestion